src/gateway.rs

```diff
@@ -1932,49 +1877,47 @@ mod events {
 #[cfg(test)]
 mod example {
     use super::*;
+    use std::sync::atomic::{AtomicI32, Ordering::Relaxed};
 
     #[derive(Debug)]
-    struct Consumer;
+    struct Consumer {
+        name: String,
+        events_received: AtomicI32,
+    }
 
     impl Observer<types::GatewayResume> for Consumer {
-        fn update(&mut self, data: &types::GatewayResume) {
-            println!("{}", data.token)
+        fn update(&self, _data: &types::GatewayResume) {
+            self.events_received.fetch_add(1, Relaxed);
```

Comment:
See, because you can't use a mutable reference you had to use an atomic value here

Reply:
This is an extension of the other comment though, see that one

Reply:
It's fine, it fits with the new observer behaviours